.github/workflows/cygwin.yml

```diff
@@ -18,6 +18,9 @@ on:
       - ".github/workflows/cygwin.yml"
       - "run*tests.py"
 
+permissions:
+  contents: read # to fetch code (actions/checkout)
```

Comment:
The reason for needing at least read is basically assumed by default so I don't think we need a comment to talk about actions/checkout.

Reply:
The idea here is to have the restrictive global permissions in case someone later adds another job. Let me know what you think and I can move it to the global.